src/main/java/edu/ucsb/cs156/frontiers/controllers/TeamsController.java

```diff
@@ -0,0 +1,368 @@
+        } else {
+          counts[rowResult.status.ordinal()]++;
+        }
+      }
+      return new TeamCreationResponse(
+          sourceType,
+          counts[TeamMemberStatus.CREATED.ordinal()],
+          counts[TeamMemberStatus.EXISTS.ordinal()],
+          failed);
+    }
+  }
+
+  /**
+   * This method returns a list of all teams for a course
+   *
+   * @return a list of all teams for a course
+   */
+  @Operation(summary = "List all teams")
+  @PreAuthorize("@CourseSecurity.hasManagePermissions(#root, #courseId)")
+  @GetMapping("/all")
+  public Iterable<Team> allTeams(@RequestParam Long courseId) {
+    Iterable<Team> teams = teamRepository.findByCourseId(courseId);
+    return teams;
+  }
+
+  /**
+   * Retrieves a list of mappings between roster students and teams for a given course. Each mapping
+   * represents a relationship between a team and its members.
+   *
+   * @param courseId the unique identifier of the course for which team mappings are retrieved
+   * @return an iterable collection of {@code TeamMemberMapping} objects representing the mappings
+   */
+  @Operation(summary = "List the mapping of Roster Students to Teams")
+  @PreAuthorize("@CourseSecurity.hasManagePermissions(#root, #courseId)")
+  @GetMapping("/mapping")
+  public Iterable<TeamMemberMapping> teamMemberMapping(@RequestParam Long courseId) {
+    List<Team> teams =
+        courseRepository
+            .findById(courseId)
+            .orElseThrow(() -> new EntityNotFoundException(Course.class, courseId))
+            .getTeams();
+    List<TeamMemberMapping> mappings = new ArrayList<>();
+    for (Team team : teams) {
+      for (TeamMember member : team.getTeamMembers()) {
+        mappings.add(TeamMemberMapping.from(member));
+      }
+    }
+    return mappings;
+  }
+
+  /**
+   * This method returns a single team by its id
+   *
+   * @param id the id of the team
+   * @return the team
+   */
+  @Operation(summary = "Get a single team")
+  @PreAuthorize("@CourseSecurity.hasManagePermissions(#root, #courseId)")
+  @GetMapping("")
+  public Team getTeamById(
+      @Parameter(name = "id") @RequestParam Long id, @RequestParam Long courseId) {
+    Team team =
+        teamRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Team.class, id));
+    return team;
+  }
+
+  /**
+   * This method deletes a team by its id
+   *
+   * @param id the id of the team to delete
+   * @return a message indicating the team was deleted
+   */
+  @Operation(summary = "Delete a team")
+  @PreAuthorize("@CourseSecurity.hasManagePermissions(#root, #courseId)")
+  @DeleteMapping("")
+  public Object deleteTeam(
+      @Parameter(name = "id") @RequestParam Long id, @RequestParam Long courseId) {
+    Team team =
+        teamRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Team.class, id));
+    teamRepository.delete(team);
+    return genericMessage("Team with id %s deleted".formatted(id));
```

Comment:
This doesn't seem to delete them from the team.  Do you need to do an additional step to delete them from the team, or perhaps adjust the cascade in some way?

Reply:
This is a must fix.  The other concern is ok to kick down the road (as long as put in a PRI-1 issue for it to fix before fall.)